src/aproxy.py

```diff
@@ -231,30 +373,40 @@ def _render_nftables_rules(self) -> str:
         }}
         """
 
-    def write_nft_config(self) -> None:
-        """Write nft config to disk."""
-        NFT_CONF_FILE.parent.mkdir(parents=True, exist_ok=True)
-        NFT_CONF_FILE.write_text(textwrap.dedent(self._render_nftables_rules()), encoding="utf-8")
+    def check_relation_availability(self) -> None:
```

Comment:
Either rename `_get_primary_ip` to `check_relation_availability` or keep the "get the IP part" in `_get_primary_ip` and move the "check relation and binding" part in this method